src/drivers/imu/murata/sch16t/SCH16T.cpp

```diff
@@ -108,9 +108,28 @@ int SCH16T::probe()
 	PX4_INFO("COMP_ID:\t 0x%0x", comp_id);
 	PX4_INFO("ASIC_ID:\t 0x%0x", asic_id);
 
-	bool success = asic_id == 0x21 && comp_id == 0x23;
+	// Determine chip version based on COMP_ID and ASIC_ID
 
-	return success ? PX4_OK : PX4_ERROR;
+	if (asic_id == 0x21 && comp_id == 0x23) {
```

Comment:
For actually storing the detected version I would do all of this in the configure() (after the reset) where it actually matters.